src/output.cpp

```diff
@@ -72,9 +72,13 @@ void ProgressBar::update(int pos)
 	if (pos == pos_prev)
 		return;
 
+#ifdef SCALEDPROGRESSBAR
```

Comment:
Remove the condtional for the on/off. Always check for the width, then set to the max if it exceeds. Something like this:
```C++
w = this->get_console_width();
#ifdef MAXPROGBARWIDTH
if (w > MAXPROGBARWIDTH)
    w = MAXPROGBARWIDTH
#endif
w -= 8;
if (w <= 0)
    return;
```